src/agent0/core/hyperdrive/policies/simple_lp.py

```diff
@@ -0,0 +1,172 @@
+    @dataclass(kw_only=True)
+    class Config(HyperdriveBasePolicy.Config):
+        """Custom config arguments for this policy."""
+
+        pnl_target: FixedPoint = FixedPoint("100")
+        """The target PNL for the bot, in base."""
+        delta_liquidity: FixedPoint = FixedPoint("100")
+        """How much liquidity to add or remove, depending on policy outcome, in base."""
+        minimum_liquidity_tokens: FixedPoint = FixedPoint("100")
+        """
+        Minimum liquidity the bot will provide.
+        It will keep this much liquidity in the pool, even if it is losing money.
+        """
+        lookback_length: int = 10
+        """
+        How many steps back to look when computing PNL progress over time.
+        Each time `action` is called is a step.
+        """
+
+    def __init__(
+        self,
+        policy_config: Config,
+    ):
+        """Initialize the bot.
+
+        Arguments
+        ---------
+        policy_config: Config
+            The custom arguments for this policy
+        """
+        super().__init__(policy_config)
+        assert policy_config.pnl_target > FixedPoint(0), "PNL target must be greater than zero."
+        self.policy_config = policy_config
+        self.pnl_history: list[tuple[FixedPoint, FixedPoint]] = []
+        self.total_base_spent: FixedPoint = FixedPoint(0)
+
+    def action(
+        self, interface: HyperdriveReadInterface, wallet: HyperdriveWallet
+    ) -> tuple[list[Trade[HyperdriveMarketAction]], bool]:
+        """Specify actions.
+
+        Arguments
+        ---------
+        interface: HyperdriveReadInterface
+            Interface for the market on which this agent will be executing trades (MarketActions).
+        wallet: HyperdriveWallet
+            The agent's wallet.
+
+        Returns
+        -------
+        tuple[list[MarketAction], bool]
+            A tuple where the first element is a list of actions,
+            and the second element defines if the agent is done trading.
+        """
+        # Get the current state of the pool & the bot's position
+        current_block = interface.get_current_block()
+        pool_state = interface.get_hyperdrive_state(current_block)
+        current_holding = wallet.lp_tokens * pool_state.pool_info.lp_share_price
+
+        # Need to be in the game to play it
+        if self.policy_config.minimum_liquidity_tokens > current_holding:
+            trade_amount = self.policy_config.minimum_liquidity_tokens - current_holding
+            return [add_liquidity_trade(trade_amount)], False
+
+        # Get current PNL
+        pnl = current_holding - self.total_base_spent
+        self.pnl_history.append((FixedPoint(interface.get_block_timestamp(current_block)), pnl))
+        # Prune history
+        if FixedPoint(len(self.pnl_history)) > self.policy_config.lookback_length:
+            self.pnl_history = self.pnl_history[-self.policy_config.lookback_length :]
+        # PNL slope is dy/dx, where y is PNL and x is time.
```

Comment:
trying to think about what this algorithm does, i think a simple average gives equal weight to each observation, even though the time passed may vary drastically. not sure what this adds over just taking dy/dx of the whole sample, which would avoid that issue, while being simpler. but we know the choice of return metric is arbitrary and has many implementations, so as a starting point, this is fine.

Reply:
I opted against using `(last_y - first_y) / (last_x - first_x)` because it puts a lot of emphasis on the first and last points. For example, if the LP had a large spike in PNL in the middle, but at the end points earned approximately the same PNL, then the slope would be ~0. I think We don't want this; we want the metric to be an average of PNL changes over the time span.

Regardless of whether we average each spline or we take the global average, the delta x (time passed) is arbitrary. However, by computing a slope, I think we are normalizing to whatever that time passed may be.

Reply:
you're right the statement doesn't hold when the segments vary in size. but the current implementation is a simple average across slopes, not weighted by segment size. so your rate of return (slope, not pnl) over a 1yr period would get the same weight as 1block period. to add in the time weighting would just be multiplying by dx... which means just not dividing by dx in the first place, so might actually be simpler and gives us pnl instead of rate of return. which at that point is the same as the aggregate, since it's just the sum of dy.

also ignoring 1 really bad trade if most of your trades are good is not a good idea. you're thinking of it as an outlier that doesn't represent the actual distribution. but some distributions are actually fat-tailed left-skewed, like selling lottery tickets or [fixed income arbitrage](https://www.johnhcochrane.com/s/Duarte_Longstaff_Yu_nickelsRFS.pdf) which is actually what we're doing here, fixed income just means rates, lovingly referred to as "picking up nickels in front of a steamroller" (you should consider the steamroller!)